File: src/main/java/com/liquor/sort/SelectSort.java
package com.liquor.sort;

import java.util.Arrays;

public class SelectSort {

    public static void main(String[] args) {
        int[] arr = new int[]{2,3,1,5,4,7,6,11};
        final int length = arr.length;
        int temp;
        for (int i = 0; i < length - 1; i++) {
           int minIndex = i;
            for (int j = i + 1; j < length; j++) {
                 if (arr[j] < arr[minIndex]) {
                     minIndex = j;
                 }
                 if (j == length - 1) {
                     temp = arr[minIndex];
                     arr[minIndex] = arr[i];
                     arr[i] = temp;
                 }
            }
        }
        System.out.println(Arrays.toString(arr));
    }
}

File: src/main/scala/com/loquor/sort/SelectSort.scala
package com.loquor.sort


object SelectSort {


  def main(args: Array[String]): Unit = {
    val arr = Array(2, 3, 1, 5, 4, 7, 6, 11)

    val length = arr.length

    var temp = -1
    for (i <- 0 until length) {
      var minIndex = i
      for (j <- i + 1 until length) {
        if (arr(minIndex) > arr(j)) {
          minIndex = j
        }

        if (j == length - 1) {
          temp = arr(minIndex)
          arr(minIndex) = arr(i)
          arr(i) = temp
        }
      }
    }
    println(arr.mkString("Array(", ", ", ")"))
  }
}
